pink/barriers/position_barrier.py

```diff
@@ -0,0 +1,150 @@
+                Defaults to None, meaning no maximum limit is applied.
+            gain (Union[float, np.ndarray], optional): regularization gain.
+                Defaults to 1.0.
+            r (float, optional): penalty weighting parameter. Defaults to 0.0.
+
+        Raises:
+            NoPositionLimitProvided: neither minimum nor maximum
+                position limit is provided.
+        """
+        indices = [0, 1, 2] if indices is None else indices
+
+        if p_min is None and p_max is None:
+            raise NoPositionLimitProvided(
+                f"Position barrier for frame {frame} requires either \
+                    p_min or p_max"
+            )
+
+        dim = 0
+        if p_min is not None:
+            dim += len(indices)
+        if p_max is not None:
+            dim += len(indices)
+
+        # If both p_min and p_max are specified, they must have
+        # the same gains repeated along the dimension
+        if isinstance(gain, np.ndarray) and len(gain) != dim:
+            gain = np.tile(gain, 2)
+
+        super().__init__(
+            dim,
+            gain=gain,
+            r=r,
+        )
+
+        self.indices = indices
+        self.frame = frame
+        self.p_min = p_min
+        self.p_max = p_max
+
+    def compute_barrier(self, configuration: Configuration) -> np.ndarray:
+        r"""Compute the value of the barrier function.
+
+        The barrier function is computed based on the position of the specified
+        frame in the world coordinate system. It considers the minimum and
+        maximum position bounds along the selected axes.
+
+        Args:
+            configuration: Robot configuration :math:`\boldsymbol{q}`.
+
+        Returns:
+            Value of the barrier function
+            :math:`\boldsymbol{h}(\boldsymbol{q})`.
+        """
+        pos_world = configuration.get_transform_frame_to_world(
+            self.frame
+        ).translation
+        barriers = []
+        if self.p_min is not None:
+            barriers.append(pos_world[self.indices] - self.p_min)
+        if self.p_max is not None:
+            barriers.append(self.p_max - pos_world[self.indices])
+
+        return np.concatenate(barriers)
+
+    def compute_jacobian(self, configuration: Configuration) -> np.ndarray:
+        r"""Compute the Jacobian matrix of the barrier function.
+
+        The Jacobian matrix is computed based on the position Jacobian of the
+        specified frame. The Jacobian is transformed to align with the world
+        coordinate system and only the selected indices are considered.
+
+        Args:
+            configuration: Robot configuration :math:`\boldsymbol{q}`.
+
+        Returns:
+            Jacobian matrix
+            :math:`\frac{\partial \boldsymbol{h}}{\partial \boldsymbol{q}}(\boldsymbol{q})`.
+        """  # noqa: E501
+        pos_jac = configuration.get_frame_jacobian(self.frame)[:3]
```

Comment:
```suggestion
        pos_jacobian_in_frame = configuration.get_frame_jacobian(self.frame)[:3]
```

Reply:
Thanks, we will checkout these towards the code